clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -784,6 +784,16 @@ template <typename T> using bind_param_t = typename bind_param<T>::type;
 class KernelObjVisitor {
   Sema &SemaRef;
 
+  template <typename ParentTy, typename... Handlers>
```

Comment:
The 'base' is now hidden in the 'Impl' function.  This makes it a little more clear what is going on I believe.